cfme/tests/infrastructure/test_providers.py

```diff
@@ -234,3 +253,46 @@ def test_infrastructure_add_provider_trailing_whitespaces():
     assert view.username.help_block == "Spaces are prohibited"
     view = prov.create_view(InfraProviderAddView)
     assert not view.add.active
+
+
+def test_infra_discovery_screen(appliance):
+    view = navigate_to(InfraProvider, 'Discover')
+    assert view.is_displayed
+    assert view.vmware.is_displayed
+    assert view.scvmm.is_displayed
+    assert view.rhevm.is_displayed
+
+    view.vmware.click()
+    assert view.vmware.selected
+    view.vmware.click()
+    assert not view.vmware.selected
+
+    view.scvmm.click()
+    assert view.scvmm.selected
+    view.scvmm.click()
+    assert not view.scvmm.selected
+
+    view.rhevm.click()
+    assert view.rhevm.selected
+    view.rhevm.click()
+    assert not view.rhevm.selected
+
+    if appliance.version >= '5.9.2':
+        view.osp_infra.click()
```

Comment:
**REQUIRED:** We should assert ```is_displayed``` like we do on the others, inside this conditional.